normandy/base/tests/api/test_api.py

```diff
@@ -235,3 +236,121 @@ def test_logged_out(self, settings):
             "logout_url": settings.OIDC_LOGOUT_URL,
             "github_url": settings.GITHUB_URL,
         }
+
+
+@pytest.mark.django_db
+class TestUserAPI(object):
+    def test_it_works(self, api_client):
+        user = User.objects.first()
+        group = GroupFactory()
+        user.groups.add(group)
+        res = api_client.get("/api/v3/user/")
+        assert res.status_code == 200
+        assert res.data == {
+            "count": 1,
+            "next": None,
+            "previous": None,
+            "results": [
+                {
+                    "id": user.id,
+                    "first_name": user.first_name,
+                    "last_name": user.last_name,
+                    "email": user.email,
+                    "groups": [{"id": group.id, "name": group.name}],
+                }
+            ],
+        }
+
+    def test_must_have_permission(self, api_client):
+        user = UserFactory(is_superuser=False)
+        api_client.force_authenticate(user=user)
+        res = api_client.get("/api/v3/user/")
+        assert res.status_code == 403
+
+        ct = ContentType.objects.get_for_model(User)
+        permission = Permission.objects.get(codename="change_user", content_type=ct)
+        user.user_permissions.add(permission)
+        user = User.objects.get(pk=user.pk)
+        api_client.force_authenticate(user=user)
+        res = api_client.get("/api/v3/user/")
+        assert res.status_code == 200
+
+    def test_cannot_delete_self(self, api_client):
+        user = User.objects.first()
+        res = api_client.delete(f"/api/v3/user/{user.id}/")
+        assert res.status_code == 403
+
+
+@pytest.mark.django_db
+class TestGroupAPI(object):
```

Comment:
I think that the model viewset also allows creating and deleting groups. Can you add tests for that? Alternatively, do we even want the ability to do that?

Reply:
I figured we wanted that ability but I can go pull it out if not. I did add tests though.

Reply:
It seems only half useful, since even if you can create groups you can't manipulate permissions. However, half useful is better than nothing, so lets keep it.